tests/ga/test_extension.py

```diff
@@ -920,39 +950,37 @@ def test_ext_handler_download_failure_permanent_ProtocolError(self, mock_add_eve
     @patch('azurelinuxagent.common.event.add_event')
-    def test_ext_handler_download_failure_permanent_with_ExtensionDownloadError_and_triggered(self, mock_add_event,
-                                                                                              mock_error_state, *args):
-        test_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE)
-        exthandlers_handler, protocol = self._create_mock(test_data, *args)
-        protocol.get_ext_handler_pkgs = Mock(side_effect=ExtensionDownloadError)
+    def test_ext_handler_download_errors_should_be_reported_only_on_new_goal_state(self, mock_add_event, *args):
 
-        mock_error_state.return_value = True
-
-        exthandlers_handler.run()
-
-        self.assertEquals(1, mock_add_event.call_count)
-        args, kw = mock_add_event.call_args_list[0]
-        self.assertEquals(False, kw['is_success'])
-        self.assertTrue("Failed to get artifact for over" in kw['message'])
-        self.assertTrue("ExtensionDownloadError" in kw['message'])
-        self.assertEquals("Download", kw['op'])
+        def _assert_mock_add_event_call(expected_call_count):
+            self.assertEquals(expected_call_count, mock_add_event.call_count, "Call count do not match")
+            args, kw = mock_add_event.call_args
+            self.assertEquals(False, kw['is_success'], "The event should have failed")
+            self.assertTrue("Failed to download artifacts" in kw['message'], "Incorrect failure, it should have failed to download")
+            self.assertTrue("ExtensionDownloadError" in kw['message'], "Incorrect error, it should be a DownloadError")
+            self.assertEquals("Download", kw['op'], "Incorrect Operation, it should be a download error")
 
-    @patch('azurelinuxagent.common.errorstate.ErrorState.is_triggered')
-    @patch('azurelinuxagent.common.event.add_event')
-    def test_ext_handler_download_failure_permanent_with_ExtensionDownloadError_and_not_triggered(self, mock_add_event,
-                                                                                                  mock_error_state,
-                                                                                                  *args):
         test_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE)
         exthandlers_handler, protocol = self._create_mock(test_data, *args)
         protocol.get_ext_handler_pkgs = Mock(side_effect=ExtensionDownloadError)
 
-        mock_error_state.return_value = False
+        exthandlers_handler.run()
+        _assert_mock_add_event_call(expected_call_count=1)
+        self._assert_handler_status(protocol.report_vm_status, "NotReady", 0, "1.0.0")
 
+        # Re-run exthandler.run without updating the GS and ensure we dont report error
         exthandlers_handler.run()
+        _assert_mock_add_event_call(expected_call_count=1)
+        self._assert_handler_status(protocol.report_vm_status, "NotReady", 0, "1.0.0")
```

Comment:
This line will fail this test currently because the cleanup logic cleans up the state and we dont report it again. Once the PR #1889  is merged, this test should pass.